docs/auxil/admonition_inserter.py

```diff
@@ -265,12 +266,54 @@ def _create_returned_in(self) -> dict[type, str]:
                 except NotImplementedError as e:
                     raise NotImplementedError(
                         f"Error generating Sphinx 'Returned in' admonition "
-                        f"(admonition_inserter.py). {cls}, method {method}. "
+                        f"(admonition_inserter.py). {cls}, method {method_name}. "
                         f"Couldn't resolve type hint in return annotation {ret_annot}. {str(e)}"
                     )
 
         return self._generate_admonitions(methods_for_class, admonition_type="returned_in")
 
+    def _create_shortcuts(self) -> dict[collections.abc.Callable, str]:
+        """Creates a dictionary with 'Shortcuts' admonitions for Bot methods that
+        have shortcuts in other classes.
+        """
+
+        # pattern for looking for calls to Bot methods only
+        bot_method_pattern = re.compile(
+            r"""\s*  # any number of whitespaces
+            (?<=return\sawait\sself\.get_bot\(\)\.)  # lookbehind
+            .+  # the method name we are looking for
+            (?=\() # lookahead: closing bracket
+            """,
+            re.VERBOSE,
+        )
+
+        # Generate a mapping of methods of classes to links to Bot methods which they are shortcuts
+        # for, i.e. {<function Bot.send_voice at ...>: {:meth:`telegram.User.send_voice`, ...}
+        shortcuts_for_bot_method = defaultdict(set)
+
+        # inspect methods of all telegram classes for return statements that indicate
+        # that this given method is a shortcut for a Bot method
+        for class_name, cls in inspect.getmembers(telegram, predicate=inspect.isclass):
+
+            for method_name, method in inspect.getmembers(cls, predicate=inspect.isfunction):
+
+                # .getsourcelines() returns a tuple. Item [1] is an int (length of line?)
+                relevant_return_lines = [
+                    line
+                    for line in inspect.getsourcelines(method)[0]
+                    if bot_method_pattern.search(line)
+                ]
+
+                for line in relevant_return_lines:
+                    bot_method_name = bot_method_pattern.search(line).group()
```

Comment:
why do we need to `search(line)` twice? and what happens with things like

```python
return await (
    selt.get_bot()
    .some_long_method_name(…)
)
```
Does the regex catch these multiline things?

Reply:
No it doesn't catch them but we don't have such things in the code :)

Maybe even our pre-commit hooks wouldn't allow this :)

Reply:
I'm rather sure that black would enforce that 😉 I guess if you `"".join(lines)` and regex-search on that, that would both cover the multiline-case as well as remove the need to loop twice

Reply:
I refactored in 659faa19e2157a0354b199f4dc8d83e9e14a9f08 to avoid using the pattern twice. Still not sure it's worth it to join the lines because we don't have any Bot names that are that long. Even this method name would fit into one line:
```py
return await self.get_bot().send_message_and_forget_all_about_it_the_very_next_morning(
```

Reply:
mh okay. let's keep it, then :D